x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/AllocationFailuresResetOnShutdownIT.java

```diff
@@ -0,0 +1,302 @@
+        var stateAfterFailures = clusterAdmin().prepareState(TEST_REQUEST_TIMEOUT).get().getState();
+        var shardAfterFailures = stateAfterFailures.routingTable().index("index1").shard(0).primaryShard();
+        assertThat(shardAfterFailures, notNullValue());
+        assertThat(shardAfterFailures.state(), equalTo(ShardRoutingState.STARTED));
+        assertThat(stateAfterFailures.nodes().get(shardAfterFailures.currentNodeId()).getName(), equalTo(node1));
+        failRelocation.set(false);
+        // A non-RESTART node shutdown should reset the counter and allow more relocation retries
+        final var request = createRequest(randomFrom(SingleNodeShutdownMetadata.Type.values()), node1, node2);
+        client().execute(PutShutdownNodeAction.INSTANCE, request).get();
+        if (request.getType() != SingleNodeShutdownMetadata.Type.RESTART) {
+            assertBusy(() -> {
+                var stateAfterNodeJoin = internalCluster().clusterService().state();
+                var relocatedShard = stateAfterNodeJoin.routingTable().index("index1").shard(0).primaryShard();
+                assertThat(relocatedShard.relocationFailureInfo().failedRelocations(), Matchers.lessThan(maxAttempts));
+                assertThat(relocatedShard, notNullValue());
+                assertThat(stateAfterNodeJoin.nodes().get(relocatedShard.currentNodeId()).getName(), not(equalTo(node1)));
+            });
+        } else {
+            var state = clusterAdmin().prepareState(TEST_REQUEST_TIMEOUT).get().getState();
+            var shard = state.routingTable().index("index1").shard(0).primaryShard();
+            assertThat(shard, notNullValue());
+            assertThat(shard.relocationFailureInfo().failedRelocations(), equalTo(maxAttempts));
+            assertThat(state.nodes().get(shard.currentNodeId()).getName(), equalTo(node1));
+        }
+    }
+
+    public void testResetRelocationFailuresOnNodeShutdownRemovalOfExistingNode() throws Exception {
+        String node1 = internalCluster().startNode();
+        createIndex("index1", 1, 0);
+        ensureGreen("index1");
+        final var failRelocation = new AtomicBoolean(true);
+        String node2 = internalCluster().startNode();
+        internalCluster().startNode();
+        internalCluster().getInstances(MockIndexEventListener.TestEventListener.class)
+            .forEach(testEventListener -> testEventListener.setNewDelegate(new IndexEventListener() {
+                @Override
+                public void beforeIndexShardCreated(ShardRouting routing, Settings indexSettings) {
+                    if (failRelocation.get()) {
+                        throw new RuntimeException("FAIL");
+                    }
+                }
+            }));
+        // add shutdown to the new node
+        final var request = createRequest(randomFrom(SingleNodeShutdownMetadata.Type.values()), node2, randomIdentifier());
+        client().execute(PutShutdownNodeAction.INSTANCE, request).get();
```

Comment:
Similar nit here. It applies to a few more places and I won't comment on all of them to avoid being too repetitive.
```suggestion
        safeGet(client().execute(PutShutdownNodeAction.INSTANCE, request));
```